src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,101 @@
+
+package frc.robot.subsystems.shooter;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import org.littletonrobotics.junction.Logger;
+
+public class Shooter extends SubsystemBase {
+
+  /** Creates a new shooter. */
+  private final ShooterIO io;
+
+  private boolean PIDMode = false;
+  private double currentVelocitySetpoint;
+
+  private SimpleMotorFeedforward shooterFeedforward =
+      new SimpleMotorFeedforward(
+          ShooterConstants.SHOOTER_KS.get(), ShooterConstants.SHOOTER_KV.get());
+  private PIDController shooterFeedack =
+      new PIDController(ShooterConstants.SHOOTER_KP.get(), 0, ShooterConstants.SHOOTER_KD.get());
+
+  private final ShooterIOInputsAutoLogged inputs = new ShooterIOInputsAutoLogged();
+
+  public Shooter(ShooterIO io) {
+    this.io = io;
+  }
+
+  public void periodic() {
+    io.updateInputs(inputs);
+    Logger.processInputs("Shooter", inputs);
+    Logger.recordOutput("Shooter/setPointVelocity", shooterFeedack.getSetpoint());
+    Logger.recordOutput("Shooter/error", shooterFeedack.getVelocityError());
+    if (Constants.tuningMode) {
+      if (ShooterConstants.SHOOTER_KS.hasChanged(hashCode())
+          || ShooterConstants.SHOOTER_KD.hasChanged(hashCode())) {
+        shooterFeedforward =
+            new SimpleMotorFeedforward(
+                ShooterConstants.SHOOTER_KS.get(), ShooterConstants.SHOOTER_KV.get());
+      }
+      if (ShooterConstants.SHOOTER_KP.hasChanged(hashCode())
+          || ShooterConstants.SHOOTER_KD.hasChanged(hashCode())) {
+        Logger.recordOutput("Shooter/newP", ShooterConstants.SHOOTER_KP.get());
+        shooterFeedack.setPID(
+            ShooterConstants.SHOOTER_KP.get(), 0, ShooterConstants.SHOOTER_KD.get());
+      }
+    }
+    if (PIDMode) {
+      io.setShooterVoltage(
+          shooterFeedack.calculate(inputs.shooterLeaderVelocityRadPerSec, currentVelocitySetpoint));
+    }
+  }
+
+  public Command stop() {
+    return Commands.run(
+        () -> {
+          io.setShooterVoltage(0.0);
+        },
+        this);
+  }
+
+  public Command shootFeedFoward(double velocitySetpoint) {
+    return Commands.run(
+        () -> io.setShooterVoltage(shooterFeedforward.calculate(currentVelocitySetpoint)));
+  }
+
+  public Command shoot(double velocitySetpoint) {
+    return Commands.run(
+        () -> {
+          PIDMode = true;
+          currentVelocitySetpoint = velocitySetpoint;
+        },
+        this);
+  }
+
+  public Command manualShoot(double volts) {
+    return Commands.run(() -> io.setShooterVoltage(volts), this);
+  }
+
+  public boolean getShooterSpeedIsReady() {
+    if (shooterFeedack.getSetpoint() == 0) {
+      return false;
+    } else {
+      return shooterFeedack.atSetpoint();
```

Comment:
make sure that in your constructor you define what the tolerance is, or this will always return true or false I forget